File: lintcode/naive/sort_integers/py/sort_integers.py
class Solution:
    # @param {int[]} A an integer array
    # @return nothing
    def sortIntegers(self, A):
        # Bubble sort.
    
        wasSwapped = False
        
        for i in xrange(len(A)):
            for j in xrange(i, len(A)):
                if A[i] > A[j]:
                    A[i], A[j] = A[j], A[i]
                    wasSwapped = True
                    
            if not wasSwapped:
                break

File: lintcode/naive/sort_integers/cpp/sort_integers.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    /**
     * @param A an integer array
     * @return void
     */
    void sortIntegers(std::vector<int>& A) 
    {
        // Bubble sort.
        
        auto wasSwapped = false;
        
        for (decltype(A.size()) i{ 0 }; i < A.size(); ++i) 
        {
            for (auto j = i; j < A.size(); ++j) 
            {
                if (A[j] < A[i]) {
                    std::swap(A[i], A[j]);
                    wasSwapped = true;
                }
            }
            
            if (!wasSwapped) break;
        }
    }
};
